From: Erik Johnston <erikj@matrix.org>
Date: Fri, 10 Jun 2016 13:24:51 +0100
Subject: Honour config.web_client

---
 synapse/app/homeserver.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/synapse/app/homeserver.py b/synapse/app/homeserver.py
index 5a0329b..492b05f 100755
--- a/synapse/app/homeserver.py
+++ b/synapse/app/homeserver.py
@@ -146,7 +146,7 @@ class SynapseHomeServer(HomeServer):
                         SERVER_KEY_V2_PREFIX: KeyApiV2Resource(self),
                     })
 
-                if name == "webclient":
+                if name == "webclient" and config.web_client:
                     resources[WEB_CLIENT_PREFIX] = build_resource_for_web_client(self)
 
                 if name == "metrics" and self.get_config().enable_metrics:
