Index: package-synapse-debian/synapse/handlers/auth.py
===================================================================
--- package-synapse-debian.orig/synapse/handlers/auth.py
+++ package-synapse-debian/synapse/handlers/auth.py
@@ -876,10 +876,8 @@ class AuthHandler(BaseHandler):
         """
 
         def _do_validate_hash():
-            return bcrypt.checkpw(
-                password.encode('utf8') + self.hs.config.password_pepper,
-                stored_hash.encode('utf8')
-            )
+            return bcrypt.hashpw(password.encode('utf8') + self.hs.config.password_pepper,
+                                 stored_hash.encode('utf8')) == stored_hash
 
         if stored_hash:
             return make_deferred_yieldable(
Index: package-synapse-debian/synapse/python_dependencies.py
===================================================================
--- package-synapse-debian.orig/synapse/python_dependencies.py
+++ package-synapse-debian/synapse/python_dependencies.py
@@ -47,7 +47,7 @@ REQUIREMENTS = {
     "pyyaml": ["yaml"],
     "pyasn1": ["pyasn1"],
     "daemonize": ["daemonize"],
-    "bcrypt": ["bcrypt>=3.1.0"],
+    "bcrypt": ["bcrypt"],
     "pillow": ["PIL"],
     "pydenticon": ["pydenticon"],
     "sortedcontainers": ["sortedcontainers"],
